File: Algorithm/baekjoon/2578_빙고.py
# baekjoon source = "https://www.acmicpc.net/problem/2578"
'''
11 12 2 24 10
16 1 13 3 25
6 20 5 21 17
19 4 8 14 9
22 15 7 23 18
5 10 7 16 2
4 22 8 17 13
3 18 1 6 25
12 19 23 14 21
11 24 9 20 15
'''
import sys

def check():
    
    cnt = 0
    for i in range(5):
        if arr[i].count(0) == 5:
            cnt += 1
    if cnt >= 3:
        return True

    for j in range(5):
        if arr[j][i] == 0:
            if j == 4:
                cnt += 1
        else:
            break
    if cnt >= 3:
        return True
    
    for j in range(5):
        if arr[j][j] == 0:
            if j == 4:
                cnt +=1
        else:
            break
    
    for j in range(5,-1,-1):
        if arr[i][j] == 0:
        
            



arr = [list(map(int,sys.stdin.readline().split())) for _ in range(5)]
check = [list(map(int,sys.stdin.readline().split())) for _ in range(5)]

for i in range(5):
    for j in range(5):
        for a in range(5):
            if check[i][j] in arr[a]:
                idx = arr[a].index(check[i][j])
                arr[a][idx] = 0
            if arr.count(0) >= 12:
                check()

File: Java/baekjoon/IM/src/_2578_빙고.java
//baekjoon source = "https://www.acmicpc.net/problem/2578"

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class _2578_ºù°í {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

		int[][] arr = new int[5][5];
		StringTokenizer st;
		for (int i = 0; i < 5; ++i) {
			st = new StringTokenizer(br.readLine());
			for (int j = 0; j < 5; ++j) {
				arr[i][j] = Integer.parseInt(st.nextToken());
			}
		}
		int[] x = new int[] { 5, 5, 5, 5, 5 };
		int[] y = new int[] { 5, 5, 5, 5, 5 };
		int[] z = new int[] { 5, 5 };
		int cnt = 0, result = 0;

		boolean total_flag = false;
		for (int i = 0; i < 5; ++i) {
			st = new StringTokenizer(br.readLine());
			boolean mid_flag = false;
			for (int j = 0; j < 5; ++j) {
				
				result++;
				int check = Integer.parseInt(st.nextToken());
				boolean flag = false;
				
				for (int a = 0; a < 5; ++a) {
					for (int b = 0; b < 5; ++b) {
						if (arr[a][b] == check) {
							x[b]--;
							if (x[b] == 0) cnt++;
							y[a]--;
							if (y[a] == 0) cnt++;
							if (a == b) {
								z[0]--;
								if (z[0] == 0) cnt++;
							}
							if (a == 0 && b == 4 || a == 1 && b == 3 || a == 2 && b == 2 || a == 3 && b == 1 || a == 4 && b == 0 ) {
								z[1]--;
								if (z[1] == 0) cnt++;
							}
							flag = true;
							if (cnt >= 3) {
								total_flag = true;
								mid_flag = true;
							}
							break;
						}
					}
					if (flag) break;
				}
				if (mid_flag) break;
			}
			if (total_flag) break;
		}
		bw.write(result+"");
		bw.flush();
	}
}
